### ISSUE TITLE
Suggestion: Encode an exact number as end marker, not unspecified value above 0FFh

### ISSUE THREAD

**Issue by ecm-pushbx** [mbaze/bzpack#7]
I believe this is the code that encodes the end marker for the default E1E1 format: https://github.com/mbaze/bzpack/blob/8627dde07d6228b3ae54dbdd283a3dce838faf75/src/Compressor.cpp#L102

In my depacker I found that you apparently encode the value `1FFh` to be used as an end marker (16 ones followed by 1 zero). I'd prefer if the value, whatever specific value you'd choose, would be fixed and dependable. This would aid my depacker's corruption detection. Accidentally encoding any value higher than `FFh` is much easier than accidentally encoding one single specific value, so the error detection is hardened if we can depend on one specific value.

**Comment by mbaze** [comment]
Yeah. Basically, this is driven by the fact that it's possible to write a single-byte test (RET C after RL C) on Z80. Basically, any value > 255 satisfies this condition. Both `1FFh` and `100h` seem to be "aesthetically pleasing" - I have no particular preference, do you?

**Comment by ecm-pushbx** [comment]
I'd chose `1FFh` out of these two simply because it frees the range `100h-1FEh` for other purposes, eg future extensions. Any larger value would though. If I understand it right though, a value such as `200h` would encode with two more bits, right?

**Comment by mbaze** [comment]
Yes. Btw, the trailing zero is not required by the assembly language decoder because even if the codeword doesn't terminate properly, the decoded number is still going to be bigger than 255. The trailing zero is only there for correctness on the C++ side of things :) But I'll try not to mess with the value, there's no particular reason to do so.

**Comment by ecm-pushbx** [comment]
I added the check to my depacker: https://hg.pushbx.org/ecm/inicomp/rev/a4c2a5a373b2
